src/routing_node.rs

```diff
@@ -692,6 +735,16 @@ impl<F> RoutingNode<F> where F: Interface {
         )
     }
 
+    fn construct_get_key_response_msg(&self, header : MessageHeader,
```

Comment:
better to pass the original header, otherwise the `construct_get_key_response_msg()` function is a bit pointless